gorp.go

```diff
@@ -604,6 +642,115 @@ func (c *ColumnMap) SetMaxSize(size int) *ColumnMap {
 	return c
 }
 
+// IndexMap represents a mapping between a Go struct field and a single
+// index in a table.
+// Unique and MaxSize only inform the
+// CreateTables() function and are not used by Insert/Update/Delete/Get.
+type IndexMap struct {
+	// Index name in db table
+	IndexName string
+
+	// If true, " unique" is added to create index statements.
+	// Not used elsewhere
+	Unique bool
+
+	// Index type supported by Dialect
+	// Postgres:  B-tree, Hash, GiST and GIN.
+	// Mysql: Btree, Hash.
+	// Sqlite: nil.
+	IndexType string
+
+	// Columns name for single and multiple indexes
+	columns []string
+}
+
+// Rename allows you to specify the index name in the table
+//
+// Example:  table.IndMap("customer_test_idx").Rename("customer_idx")
+//
+func (idx *IndexMap) Rename(indname string) *IndexMap {
+	idx.IndexName = indname
+	return idx
+}
+
+// SetUnique adds "unique" to the create index statements for this
+// index, if b is true.
+func (idx *IndexMap) SetUnique(b bool) *IndexMap {
+	idx.Unique = b
+	return idx
+}
+
+// SetIndexType specifies the index type supported by chousen SQL Dialect
+func (idx *IndexMap) SetIndexType(indtype string) *IndexMap {
+	idx.IndexType = indtype
+	return idx
+}
+
+func (m *DbMap) CreateIndex() error {
+
+	var err error
+	dialect := reflect.TypeOf(m.Dialect)
+	for _, table := range m.tables {
+		for _, index := range table.indexes {
+
+			s := bytes.Buffer{}
+			s.WriteString("create")
+			if index.Unique {
+				s.WriteString(" unique")
+			}
+			s.WriteString(" index")
+			s.WriteString(fmt.Sprintf(" %s on %s", index.IndexName,
+				table.TableName))
+			if dname := dialect.Name(); dname == "PostgresDialect" && index.IndexType != "" {
+				s.WriteString(fmt.Sprintf(" %s %s", m.Dialect.CreateIndexSuffix(), index.IndexType))
+			}
```

Comment:
As @coopernurse mentioned in https://github.com/go-gorp/gorp/pull/97#issuecomment-30667564
The Dialect's name is inspected here, this behavior should be moved to the Dialect implementations. The whole purpose of the Dialect is to handle the different in SQL Syntaxes.